### PR TITLE
Docs for PipeTransform and PipeOnDestroy interfaces

### PR DESCRIPTION
**Pull request by jeffbcross**

### CODE REVIEW

File: modules/angular2/src/core/change_detection/pipe_transform.ts [hunk 1]
```diff
@@ -1,35 +1,36 @@
...
+ * as the first argument and any parameters in the `args` Array.
  *
  * ## Syntax
  *
- * `subject | pipeName[:arg0[:arg1...]]`
+ * `value | pipeName[:arg0[:arg1...]]`
  *
  * ## Example
  *
  * The `RepeatPipe` below repeats the subject as many times as indicated by the first argument:
```
Comment (pkozlowski-opensource) [inline modules/angular2/src/core/change_detection/pipe_transform.ts]: Probably would be good to change "repeats the subject" to "repeats the value" to be consistent with other changes.

File: modules/angular2/src/core/change_detection/pipe_transform.ts [hunk 2]
```diff
@@ -37,22 +38,46 @@ export interface PipeTransform { transform(value: any, args: any[]): any; }
 /**
  * An interface that stateful pipes should implement.
  *
- * #Example
+ * A stateful pipe may produce different output, given the same input. It is
+ * likely that a stateful pipe may contain state that should be cleaned up when
+ * a binding is destroyed. For example, an observable subscription may need to
```
Comment (pkozlowski-opensource) [inline modules/angular2/src/core/change_detection/pipe_transform.ts]: people new to Rx might freak out at "an observable subscription" :-) Maybe we should drop it or re-word?

File: modules/angular2/src/core/change_detection/pipe_transform.ts [hunk 3]
```diff
@@ -37,22 +38,46 @@ export interface PipeTransform { transform(value: any, args: any[]): any; }
...
- *    this.connection.release();
- *  }
- *
- *  transform(value, args = []) {
- *    this.connection = createConnection();
- *    // ...
- *    return someValue;
- *  }
+ * @Pipe({name: 'square'})
+ * class SquarePipe implements PipeTransform, PipeOnDestroy {
```
Comment (pkozlowski-opensource) [inline modules/angular2/src/core/change_detection/pipe_transform.ts]: Maybe a simpler example (counter?) would do the trick here?

File: modules/angular2/src/core/change_detection/pipe_transform.ts (hunk 2 above)
Comment (jeffbcross) [inline modules/angular2/src/core/change_detection/pipe_transform.ts]: Hmm, subscription is such a perfect illustration of the need here though. Interval probably isn't that common of a use case, so I don't want that to be the primary use case. I could go for something more generic like "stream subscription"

File: modules/angular2/src/core/change_detection/pipe_transform.ts (hunk 1 above)
Comment (jeffbcross) [inline modules/angular2/src/core/change_detection/pipe_transform.ts]: Good catch!

File: modules/angular2/src/core/change_detection/pipe_transform.ts (hunk 2 above)
Comment (pkozlowski-opensource) [inline modules/angular2/src/core/change_detection/pipe_transform.ts]: Yep, totally agree that it is a perfect example! Maybe it just need some re-wording to make it more approachable?

File: modules/angular2/src/core/change_detection/pipe_transform.ts [hunk 4]
```diff
@@ -1,35 +1,36 @@
 /**
- * To create a Pipe, you must implement this interface.
+ * An interface that all pipes must implement.
```
Comment (vicb) [inline modules/angular2/src/core/change_detection/pipe_transform.ts]: I think we did the opposite change while reviewing with @naomiblack  (to have an active sentence)

File: modules/angular2/src/core/change_detection/pipe_transform.ts [hunk 5]
```diff
@@ -37,22 +38,50 @@ export interface PipeTransform { transform(value: any, args: any[]): any; }
...
+ * @Pipe({name: 'countdown'})
+ * class CountDown implements PipeTransform, PipeOnDestroy {
+ *   remainingTime:Number;
+ *   interval:SetInterval;
+ *   onDestroy() {
+ *     if (this.interval) {
+ *       clearInterval(this.interval);
+ *     }
+ *   }
+ *   transform(remainingTime, args = []) {
```
Comment (vicb) [inline modules/angular2/src/core/change_detection/pipe_transform.ts]: ` transform(value: any, args: any[] = [])` tom match the doc above

**Comment by jeffbcross** [comment]
Merged at https://github.com/angular/angular/commit/d276370ba148f3487c7952f180c8b8907b650cc7

**Pull request closed without merging**